scripts/vendor-oasis-core/patches/v22.2.11/assorted.patch

```diff
@@ -0,0 +1,240 @@
++	// registration.
++	RegisterEntitySignatureContext = original.RegisterEntitySignatureContext
++
++	// RegisterGenesisEntitySignatureContext is the context used for
++	// entity registration in the genesis document.
++	//
++	// Note: This is identical to non-gensis registrations to support
++	// migrating existing registrations into a new genesis document.
++	RegisterGenesisEntitySignatureContext = RegisterEntitySignatureContext
++
++	// RegisterNodeSignatureContext is the context used for node
++	// registration.
++	RegisterNodeSignatureContext = original.RegisterNodeSignatureContext
++)
+ 
+ // Backend is a registry implementation.
+ // removed interface
+diff --git a/coreapi/v22.2.11/roothash/api/block/header.go b/coreapi/v22.2.11/roothash/api/block/header.go
+index d7803c44..b364d89a 100644
+--- a/coreapi/v22.2.11/roothash/api/block/header.go
++++ b/coreapi/v22.2.11/roothash/api/block/header.go
+@@ -106,7 +106,9 @@ type Header struct { // nolint: maligned
+ // removed func
+ 
+ // EncodedHash returns the encoded cryptographic hash of the header.
+-// removed func
++func (h *Header) EncodedHash() hash.Hash {
++	return hash.NewFrom(h)
++}
+ 
+ // StorageRoots returns the storage roots contained in this header.
+ // removed func
+diff --git a/coreapi/v22.2.11/runtime/client/api/api.go b/coreapi/v22.2.11/runtime/client/api/api.go
+index ab839929..b487ded8 100644
+--- a/coreapi/v22.2.11/runtime/client/api/api.go
++++ b/coreapi/v22.2.11/runtime/client/api/api.go
+@@ -4,7 +4,6 @@ import (
+ 	roothash "github.com/oasisprotocol/nexus/coreapi/v22.2.11/roothash/api"
+ 	"github.com/oasisprotocol/oasis-core/go/common"
+ 	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
+-	"github.com/oasisprotocol/oasis-core/go/runtime/host/protocol"
+ )
+ 
+ const (
+@@ -42,7 +41,16 @@ type SubmitTxMetaResponse struct {
+ 	BatchOrder uint32 `json:"batch_order,omitempty"`
+ 
+ 	// CheckTxError is the CheckTx error in case transaction failed the transaction check.
+-	CheckTxError *protocol.Error `json:"check_tx_error,omitempty"`
++	CheckTxError RuntimeHostError `json:"check_tx_error,omitempty"`
++}
++
++// RuntimeHostError is a message body representing an error.
++// NOTE: RENAMED from "Error" and imported from github.com/oasisprotocol/oasis-core/go/runtime/host/protocol
++// as a manual step when vendoring oasis-core v22.2.11.
++type RuntimeHostError struct {
++	Module  string `json:"module,omitempty"`
++	Code    uint32 `json:"code,omitempty"`
++	Message string `json:"message,omitempty"`
+ }
+ 
+ // CheckTxRequest is a CheckTx request.
+diff --git a/coreapi/v22.2.11/staking/api/address.go b/coreapi/v22.2.11/staking/api/address.go
+index 1f706bbe..ea721dce 100644
+--- a/coreapi/v22.2.11/staking/api/address.go
++++ b/coreapi/v22.2.11/staking/api/address.go
+@@ -1,7 +1,9 @@
+ package api
+ 
+ import (
+-        original "github.com/oasisprotocol/oasis-core/go/staking/api"
++	original "github.com/oasisprotocol/oasis-core/go/staking/api"
```

Comment:
just curious - do we not vendor these types because they're unlikely to change in future oasis-core versions?

Reply:
Yeah, that's the hope anyway. If they do change, we can always go back and vendor them fully. (Just like I did in this PR with `node` -- exact same situation.)

Main motivation:
Because these types are so commonly used, it's very convenient to have them aliased to the exact same type across all our vendored copies of oasis-core. This way, Go is happier to perform automatic casts from e.g. the cobalt version of MyTypeThatContainsAddress to the damask version of MyTypeThatContainsAddress.